app/src/components/ToggleSwitch.jsx

```diff
@@ -10,21 +10,33 @@ class ToggleSwitch extends React.Component {
     super(props);
 
     this.state = {
-      checked: this.props.checked || false,
+      checked: props.checked != null ? props.checked : false,
     };
   }
 
+  componentWillReceiveProps(props){
+    this.setState({
+      checked: props.checked != null ? props.checked : this.state.checked,
+    });
+  }
+
   handleToggle(e, a) {
     this.setState({
-      checked: !this.state.checked
+      checked: !this.state.checked,
```

Comment:
I don't think this component needs a state anymore. But if it works we can change it later.